dpar/src/features/input_layers.rs

```diff
@@ -227,6 +271,89 @@ impl InputVectorizer {
             }
         }
     }
+
+    /// Vectorize a parser state into the given association measure slices.
+    ///
+    /// Add to `non_lookup_slices` the association measure between all parser state addresses
+    /// undergoing attachment.
+    pub fn realize_into_assoc_strengths(
+        &self,
+        state: &ParserState,
+        non_lookup_slices: &mut [f32],
+        attachment_addrs: &[AttachmentAddr],
+    ) {
+        for (addr, non_lookup_slice) in attachment_addrs.iter().zip(non_lookup_slices.iter_mut()) {
+            let addr_head = addr::AddressedValue {
+                address: vec![addr.head],
+                layer: addr::Layer::Token,
+            };
+            let addr_dependent = addr::AddressedValue {
+                address: vec![addr.dependent],
+                layer: addr::Layer::Token,
+            };
+
+            let head = addr_head.get(state);
+            let dependent = addr_dependent.get(state);
+
+            if head.is_some()
+                && dependent.is_some()
+                && self.layer_lookups.layer_lookup(Layer::DepRel).is_some()
+            {
+                let head = head.unwrap().to_owned();
+                let dependent = dependent.unwrap().to_owned();
+
+                let mut deprels = Vec::new();
+                self.layer_lookups
+                    .layer_lookup(Layer::DepRel)
+                    .unwrap()
+                    .lookup_values(&mut deprels);
+
+                for deprel in deprels {
```

Comment:
This is overwriting the same element of `non_lookup_slices` over and over again.

Reply:
It seems that the unit test is incomplete, since it does not catch these errors?

Reply:
Probably cause there is only one dependency relation in the test which doesn't trigger overwriting. I will add a test with more than 1 deprel.